lib/API/JSON/v1/Subscribers.php

```diff
@@ -489,21 +485,46 @@ public function sendConfirmationEmail($data = []) {
   }
 
   public function bulkAction($data = []) {
-    try {
-      if (!isset($data['listing']['filter']['segment'])) {
-        return $this->successResponse(
-          null,
-          $this->bulkActionController->apply('\MailPoet\Models\Subscriber', $data)
-        );
-      } else {
-        $bulkAction = new BulkAction($data);
-        return $this->successResponse(null, $bulkAction->apply());
+    $definition = $this->listingHandler->getListingDefinition($data['listing']);
+    $ids = $this->subscriberListingRepository->getActionableIds($definition);
+
+    $count = 0;
+    $segment = null;
+    if (isset($data['segment_id'])) {
+      $segment = $this->getSegment($data);
+      if (!$segment) {
```

Comment:
I can't remove the instance checks in the following lines because I wouldn't be sure then that segment will be defined. The block with isset hasn't to execute and then the method which needs SegmentEntity will end with an error.

Phpstan says to me too that it's a problem.

Reply:
Ahh, I see! It is a different level. So let's keep it as is.